lua/oil/init.lua

```diff
@@ -643,6 +652,9 @@ M.select = function(opts, callback)
         M.close()
       end)
     end
+    if keep_preview_alive then
+      M.select({preview = true})
+    end
```

Comment:
One of the difficulties with this is oil loads asynchronously, so it's entirely possible for us to be in the buffer by this point but the buffer hasn't rendered yet. This is a tricky situation, so I added a utility method `util.run_after_load` that will run a function once the oil buffer is ready for use. Try using that here, with some checks to make sure we're still in the buffer and that there is an entry under the cursor (otherwise we'll display an error in an empty folder). You'll want to mostly copy the logic from here

https://github.com/stevearc/oil.nvim/blob/c4cc8240f1c71defcb67c45da96e44b968d29e5f/lua/oil/view.lua#L411-L422

Reply:
Thank you!